### PR TITLE
Implementing Authentication in Rails

### PR DESCRIPTION
**Pull request by Zubenna**
## In this pull request, we did the following;
- Set up the required database
- Installed devise and configured routes, models, and controllers.
- Added members-only functionality in the app.
- Test app functionality.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -15,6 +16,7 @@
       <tr>
         <td><%= post.title %></td>
 	<td><%= post.content %></td>
+        <td><%= user_signed_in? ? post.user.username : "Anonymous" %></td>
```
Comment (serdg0) [inline app/views/posts/index.html.erb:19]: As the pathwright says, there shouldn't be any logic in the views:


Check this resources for more about it:
[Where do I put my code?](http://codefol.io/posts/Where-Do-I-Put-My-Code/)
[The beginner's guide to rails helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

File: config/routes.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
-  resources :posts
+  devise_for :users, :controllers => { registrations: 'registrations' }
+    resources :posts
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply config/routes.rb:3]: Updated method, allow only new, create and index, and modified the controller

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -9,24 +10,20 @@ def index
...
   # GET /posts/1.json
-  def show
-  end
+  def show; end
 
   # GET /posts/new
   def new
-    @post = Post.new
-    
+    @post = current_user.posts.build
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply app/controllers/posts_controller.rb:17]: is initializing an empty post, only passing the user id as a parameter

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
 
   # GET /posts
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply app/controllers/posts_controller.rb:5]: Removed autogenerated comments

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (AdfPizarro) [reply app/views/posts/index.html.erb:19]: Moved the logic to the controller to modify the active record before return it

File: README.md [hunk 5]
```diff
@@ -1,24 +1,62 @@
...
-* System dependencies
+## How to Test the Project
 
-* Configuration
+Get a local copy of the file  clone with HTTPS 
 
-* Database creation
+```
+https://github.com/AdfPizarro/members-only.git  
+```
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply README.md:19]: Added the line

**Pull request merged**